### DISCUSSION TITLE
How to debug app issues?

### DISCUSSION THREAD

**Discussion by shalak** [linuxserver/Heimdall#1133] [Q&A]
I've added portainer app and even though the access test says "Successfully communicated with the API", I'm not seeing any stats in the icon.

I edited the `www/.env` file and set the `APP_DEBUG=true`, then restarted the heimdall container, but no additional logs show up.

I only see the following `.log` files:
```
config/log/nginx/access.log
config/log/nginx/error.log
config/log/php/error.log
config/www/logs/laravel-2023-03-18.log
```
Now, the `./config/log/nginx/error.log` is empty, `./config/log/nginx/access.log` contains only access into to the heimdall page itself, `config/log/php/error.log` says `ready to handle connections` and nothing else, and lastly, the `laravel-2023-03-18.log` ends on log related to the app installation:

```
[2023-03-18 12:13:23] local.DEBUG: Download triggered for stdClass Object
(
    [appid] => 07b796b3a10eaa8b6f485ed2379187a39400b459
    [name] => Portainer
    [website] => https://portainer.io
    [license] => zlib License
    [description] => Portainer is a simple management solution for Docker. Easily manage your Docker hosts and Docker Swarm clusters via Portainer web user interface.
    [enhanced] => 1
    [tile_background] => dark
    [icon] => portainer.svg
    [sha] => 94a81b07f372894d0b6db212e2bb33681a1c839c
    [class] => \App\SupportedApps\Portainer\Portainer
)
```

How can I debug the actual API calls made by heimdall?

**Comment by shalak** [comment]
Huh, I forgot to enable the API integration, now it works correctly, but still - for future references, how do I debug the issues?

**Reply by deviantintegral** [reply]
It looks like the laravel logs are stored in `/app/www/storage/logs` which isn't exposed as a docker volume mapping. I did a basic test with that, but it broke the entire app with `/app/www/artisan` no longer existing. I think the simplest way right now is to exec into the container and tail those logs there.
